extensions/github1s/src/views/commit-list-view.ts

```diff
@@ -0,0 +1,124 @@
+/**
+ * @file GitHub Commit List View
+ * @author netcon
+ */
+
+import * as vscode from 'vscode';
+import * as queryString from 'query-string';
+import { relativeTimeTo } from '@/helpers/date';
+import repository, { RepositoryCommit } from '@/repository';
+import { GitHub1sSourceControlDecorationProvider } from '@/providers/sourceControlDecorationProvider';
+import {
+	getChangedFileCommand,
+	getCommitChangedFiles,
+} from '@/source-control/changes';
+import router from '@/router';
+
+const getCommitTreeItemDescription = (commit: RepositoryCommit): string => {
+	return [
+		commit.sha.slice(0, 7),
+		commit.commit.author.name,
+		relativeTimeTo(commit.commit.author.date),
+	].join(', ');
+};
+
+export interface CommitTreeItem extends vscode.TreeItem {
+	commit: RepositoryCommit;
+}
+
+export class CommitTreeDataProvider
+	implements vscode.TreeDataProvider<vscode.TreeItem> {
+	public static viewType = 'github1s.views.commit-list';
+
+	private forceUpdate = false;
+	private _onDidChangeTreeData = new vscode.EventEmitter<undefined>();
+	readonly onDidChangeTreeData = this._onDidChangeTreeData.event;
+
+	public updateTree() {
+		this.forceUpdate = true;
+		this._onDidChangeTreeData.fire(undefined);
+	}
+
+	async getCommitItems(): Promise<CommitTreeItem[]> {
+		const { ref } = await router.getState();
+		// only recent 100 commits will be list here
+		// TODO: implement pagination
```

Comment:
Do you have any idea of how we could support pagination in the future?

I think one possible way is to add an option at the bottom to `load more commits` and it triggers another call for the next page and update the quick pick list.

Reply:
Yes, I agree it, I also want add a special item named `Load more` below these items~